test/4.public.UrlChecker.js

```diff
@@ -139,9 +139,83 @@ describe("PUBLIC -- UrlChecker", function()
 			}).enqueue( conn.absoluteUrl );
 		});
 	});
-	
-	
-	
+
+
+
+	describe("caching", function()
+	{
+		var options;
+		beforeEach(function()
+		{
+			options = utils.options();
+			options.cacheResponses = true;
+		});
+
+		it("should check unique url only once", function(done)
+		{
+			var results = [];
+			
+
+			var instance = new UrlChecker( options,
+			{
+				link: function(result, customData)
+				{
+					results.push(result);
```

Comment:
Should use `results[customData.index] = result;` ([in context](https://github.com/stevenvachon/broken-link-checker/blob/master/test/4.public.UrlChecker.js#L186)) to avoid a test failure if results are returned in different order. Unlikely since `maxSocketsPerHost===1`, but it's precautionary.